modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/upgrade/BlogsServiceUpgrade.java

```diff
@@ -95,12 +101,55 @@ public void register(Registry registry) {
 			new com.liferay.blogs.internal.upgrade.v2_1_1.UpgradeBlogsEntry());
 	}
 
+	private UnsafeFunction<String, Boolean, Exception>
+		_getUpgradeDiscussionSubscriptionClassNameUnsafeFunction() {
+
+		return className -> {
+			DynamicQuery groupDynamicQuery = _groupLocalService.dynamicQuery();
+
+			groupDynamicQuery.add(RestrictionsFactoryUtil.eq("site", true));
```

Comment:
@IstvanD @robertoDiaz 

This is not the most important but to improve the performance of this, we can use the index:
create index IX_CFE2671B on Group_ (companyId, site, ctCollectionId);

This will help to reduce the list too.